tests/drives_tests.cpp

```diff
@@ -88,6 +98,19 @@ TEST(WildFileCmp, QuestionMark)
     EXPECT_EQ(true, WildFileCmp("TEST", "???T.???"));
 }
 
+TEST(WildFileCmp, LongCompare)
```

Comment:
@Wengier , it looks like the WildFileCmp `long_compare==true` scenario also handles the 8.3 case. 
If the LFN scenario can handle the short case, maybe we can make the WildFileCmp always handle the long scenario and get rid of all the `long_compare` branches in that function? (this is more a question than asking.. maybe WildFileCmp must stricting have 8.3 for other uses in DOSBox?)

Reply:
@kcgen While the semantics of \* and ? wildcards never change, there are difference between the actual behaviors of \* in LFN and strict 8.3 name modes, in that the latter case is less flexible - everything after * and before . is ignored in strict 8.3 name mode. For example:

In strict 8.3 name mode, \*w\* is equivalent to \*, so \*w\* will always match all files.
In LFN mode, \*w\* will match all files containing the letter "w", which is clearly more flexible.

For backward compatibility purposes they are made separate.